Investigate/fix test_search_change_column

We need to investigate/fix test_search_change_column (at least as run against prod, in-house Grid):

http://selenium.qa.mtv2.mozilla.com:8080/view/Socorro/job/socorro.prod/lastCompletedBuild/testReport/tests.test_search/TestSearchForIdOrSignature/test_search_change_column/history/

16:49:32 =================================== FAILURES ===================================
16:49:32 _____________ TestSearchForIdOrSignature.test_search_change_column _____________
16:49:32 [gw3] darwin -- Python 2.7.5 /Users/Shared/Jenkins/Home/jobs/socorro.prod/workspace/.env/bin/python
16:49:32 self = <tests.test_search.TestSearchForIdOrSignature instance at 0x103268b00>
16:49:32 mozwebqa = <pytest_mozwebqa.pytest_mozwebqa.TestSetup instance at 0x101febdd0>
16:49:32 
16:49:32     @pytest.mark.nondestructive
16:49:32     def test_search_change_column(self, mozwebqa):
16:49:32         csp = CrashStatsHomePage(mozwebqa)
16:49:32         cs_super = csp.header.click_super_search()
16:49:32         cs_super.select_field('product')
16:49:32         cs_super.select_operator('has terms')
16:49:32     
16:49:32         cs_super.click_search()
16:49:32         Assert.true(cs_super.are_search_results_found)
16:49:32         cs_super.click_more_options()
16:49:32     
16:49:32         # Delete all columns except the last one
16:49:32         for column in cs_super.columns[:-1]:
16:49:32             cs_super.click_crash_reports_tab()
16:49:32             current_column = column.column_name
16:49:32             Assert.true(current_column in cs_super.search_results_table_header.table_column_names)
16:49:32     
16:49:32             number_of_columns = len(cs_super.columns)
16:49:32             column.delete_column()
16:49:32             cs_super.wait_for_column_deleted(number_of_columns - 1)
16:49:32             Assert.false(cs_super.is_column_in_list(current_column))
16:49:32     
16:49:32             cs_super.click_search()
16:49:32             if len(cs_super.columns) > 1:
16:49:32                 cs_super.click_crash_reports_tab()
16:49:32                 Assert.true(cs_super.are_search_results_found)
16:49:32 >               Assert.false(current_column in cs_super.search_results_table_header.table_column_names)
16:49:32 
16:49:32 tests/test_search.py:104: 
16:49:32 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
16:49:32 
16:49:32 self = <class 'unittestzero.Assert'>, first = True, msg = None
16:49:32 
16:49:32     @classmethod
16:49:32     def false(self, first, msg=None):
16:49:32         """
16:49:32             Asserts that what is given is equal to False
16:49:32     
16:49:32             :Args:
16:49:32              - First object to be tested
16:49:32              - Message that will be printed if it fails
16:49:32             """
16:49:32     
16:49:32 >       assert bool(first) is False, msg
16:49:32 E       AssertionError: None
16:49:32 
16:49:32 .env/lib/python2.7/site-packages/unittestzero.py:91: AssertionError
16:49:32 ------------------------------- pytest-mozwebqa --------------------------------
16:49:32 Failing URL: https://crash-stats.mozilla.com/search/?product=Firefox&_facets=signature&_columns=signature&_columns=product&_columns=version&_columns=build_id&_columns=platform#crash-reports
16:49:32 ----------------------- generated xml file: results.xml ------------------------
16:49:32 =========================== short test summary info ============================
16:49:32 FAIL tests/test_search.py::TestSearchForIdOrSignature::()::test_search_change_column

--- a/pages/super_search_page.py
+++ b/pages/super_search_page.py
@@ -129,16 +129,16 @@ class CrashStatsSuperSearch(CrashStatsBasePage):
 
     class SearchResultHeader(Page):
 
-        _root_locator = (By.CSS_SELECTOR, '#reports-list thead')
-        _table_header_name_locator = (By.CSS_SELECTOR, 'th')
-
-        def __init__(self, testsetup):
-            Page.__init__(self, testsetup)
-            self._root_element = self.selenium.find_element(*self._root_locator)
+        _table_header_name_locator = (By.CSS_SELECTOR, '#reports-list thead th')
 
         @property
         def table_column_names(self):
-            return [table_column.text.lower() for table_column in self._root_element.find_elements(*self._table_header_name_locator)]
+            return [table_column.text.lower() for table_column in self.selenium.find_elements(*self._table_header_name_locator)]
+
+        def is_column_not_present(self, column_name):
+            WebDriverWait(self.selenium, self.timeout).until(
+                lambda s: column_name not in self.table_column_names, message='Column %s found in table header.' % column_name)
+            return True
 
     class Column(Page):
 

--- a/tests/test_search.py
+++ b/tests/test_search.py
@@ -101,7 +101,7 @@ class TestSearchForIdOrSignature:
             if len(cs_super.columns) > 1:
                 cs_super.click_crash_reports_tab()
                 Assert.true(cs_super.are_search_results_found)
-                Assert.false(current_column in cs_super.search_results_table_header.table_column_names)
+                Assert.true(cs_super.search_results_table_header.is_column_not_present(current_column))
 
         Assert.true(cs_super.columns[0].column_name in cs_super.search_results_table_header.table_column_names)
 

